src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -691,23 +695,23 @@ internal static bool Dispatch()
                     //
                     // Execute the workitem outside of any finally blocks, so that it can be aborted if needed.
                     //
+#if FEATURE_OBJCWRAPPERS
+                    if (ThreadPool.EnableDispatchAutoReleasePool)
+                    {
+#pragma warning disable CA1416 // This call site is reachable on all platforms. 'Wrappers.DrainAutoreleasePool(IntPtr)' is only supported on: 'macos'. This code is only compiled on macos.
```

Comment:
You should be able to `Debug.Assert(OperatingSystem.IsMacOs());` and then you won't need the pragma or the comment.